### ISSUE TITLE
Bindings on CommandBar not working

### ISSUE THREAD

**Issue by DVDPT** [MyToolkit/MyToolkit#23]
After I changed to the MyToolkit MtPage my binding stop working, when checking the DataContext of the buttons they are all null, and even setting the DataContext directly on the CommandBar doesn't seam to fix.

**Comment by rsuter** [comment]
Do you set the DataContext directly on the MtPage DataContext property?

**Comment by DVDPT** [comment]
Yes

**Comment by rsuter** [comment]
Ok, automatically copying the DataContext from the MtPage to the DataContext of the BottomAppBar and TopAppBar should work then?

**Comment by DVDPT** [comment]
Something strange is happening
Even using a relative binding on the command bar ( DataContext="{Binding DataContext,ElementName=RootGrid}") it also doesn't work and should.
Just setting the DataContext on the CommandBar also doesn't work. The icons seam to never receive the update

**Comment by rsuter** [comment]
The problem is that internally the class uses the original Page class and the app bars are "copied" over... Currently the resources from the MtPage are also copied to the app bars... Currently the datacontext is not copied and thus the bindings do not work... I think relative bindings will never work. I have to reproduce the problem so that i can fix this. Will inform you as soon as i have more information...

**Comment by DVDPT** [comment]
Sure until then the workaround the what I posted earlier. Not pretty but works for now.

**Comment by rsuter** [comment]
What is your working workaround?

**Comment by DVDPT** [comment]
Manually setting the datacontext on each button
```
foreach (AppBarButton source in AppBar.PrimaryCommands.Union(AppBar.SecondaryCommands))
            {
                source.DataContext = pageRoot.DataContext;
            }
```

**Comment by rsuter** [comment]
Maybe this is not working at all?
http://stackoverflow.com/questions/25753649/how-to-set-appbar-datacontext
Is it working when setting the DataContext of the default Page class?

**Comment by DVDPT** [comment]
Hum probably.
But before I switched to the MtPages I was using the regular binding and everything worked fine.

**Comment by rsuter** [comment]
Should now be fixed in NuGet version 2.5.4, please update and report if it works...

**Comment by DVDPT** [comment]
I confirm, you can close the issue
